### PR TITLE
cadl dpg, add test base

### PR DESCRIPTION
**Pull request by XiaofeiCao**
issue: https://github.com/Azure/autorest.java/issues/1946

Always generate TestBase under `generated` package for dpg.

### CODE REVIEW

File: cadl-extension/src/main/java/com/azure/cadl/model/EmitterOptions.java [hunk 1]
```diff
@@ -16,18 +16,27 @@ public class EmitterOptions {
...
     @JsonProperty(value="partial-update")
     private Boolean partialUpdate;
 
     @JsonProperty(value="service-versions")
     private List<String> serviceVersions;
 
+    @JsonProperty(value = "generate-tests")
+    private Boolean generateTests;
+
+    @JsonProperty(value = "generate-examples")
```
Comment (weidongxu-microsoft) [inline cadl-extension/src/main/java/com/azure/cadl/model/EmitterOptions.java:37]: generate-samples ?

File: javagen/src/main/java/com/azure/autorest/Javagen.java [hunk 2]
```diff
@@ -214,15 +214,15 @@ JavaPackage writeToTemplates(CodeModel codeModel, Client client, JavaSettings se
         }
 
         // Test
-        if (settings.isDataPlaneClient() && settings.isGenerateTests()) {
+        if (settings.isDataPlaneClient()) {
```
Comment (weidongxu-microsoft) [inline javagen/src/main/java/com/azure/autorest/Javagen.java:217]: I think `settings.isGenerateTests()` still needed?

I mean, default `generate-samples` and `generate-tests` to true in CadlPlugin or EmitterOptions.

User can still disable by explicitly setting it to false.

File: cadl-tests/pom.xml [hunk 3]
```diff
@@ -66,5 +66,17 @@
...
+    <dependency>
+      <groupId>com.azure</groupId>
+      <artifactId>azure-core-test</artifactId>
+      <version>1.14.1</version>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>com.azure</groupId>
+      <artifactId>azure-identity</artifactId>
+      <version>1.7.3</version>
```
Comment (weidongxu-microsoft) [inline cadl-tests/pom.xml:78]: use 1.8.0

File: cadl-extension/src/main/java/com/azure/cadl/model/EmitterOptions.java (hunk 1 above)
Comment (XiaofeiCao) [reply cadl-extension/src/main/java/com/azure/cadl/model/EmitterOptions.java:37]: Right, thanks...

File: javagen/src/main/java/com/azure/autorest/Javagen.java (hunk 2 above)
Comment (XiaofeiCao) [reply javagen/src/main/java/com/azure/autorest/Javagen.java:217]: You mean if user sets `generate-tests` to false, then the TestBase won't be generated as well? Makes sense, will do that.

Comment (weidongxu-microsoft) [reply javagen/src/main/java/com/azure/autorest/Javagen.java:217]: Yes, `TestBase` still follow the `generate-tests`. The change is when there is no tests, still generate the TestBase (so dev can add a new test class extends it, avoid writing these interceptorManager).

**Review by weidongxu-microsoft — APPROVED** [review]

**Pull request merged**